.github/workflows/update_apps.yml

```diff
@@ -0,0 +1,79 @@
+name: Update_Apps
+
+# Controls when the workflow will run
+on:  
+  # Allows you to run this workflow manually from the Actions tab
+  workflow_dispatch:
+
+# A workflow run is made up of one or more jobs that can run sequentially or in parallel
+jobs:
+  update-apps:
+    # The type of runner that the job will run on
+    runs-on: ubuntu-latest
+
+    # Steps represent a sequence of tasks that will be executed as part of the job
+    steps:
+      # Checks-out your repository under $GITHUB_WORKSPACE, so your job can access it
+      - uses: actions/checkout@v2
+
+      # Runs a set of commands using the runners shell
+      - name: Run app update scripts
+        run: |
+          #source pi-apps functions
+
+          #export all functions and variables
+          set -a
+          #make DIRECTORY equal to GITHUB_WORKSPACE, for subscripts and api functions
+          DIRECTORY=$GITHUB_WORKSPACE
+          source $GITHUB_WORKSPACE/api
+
+          #add special functions
+          get_release() {
+            curl --silent "https://api.github.com/repos/$1/releases/latest" | jq -r '.tag_name' | sed s/v//g
+          }
+
+          function validate_url(){
+            if [[ `wget -S --spider $1  2>&1 | grep 'HTTP/1.1 200 OK'` ]]; then
```

Comment:
```suggestion
            if [ "$(curl -s -o /dev/null -I -w "%{http_code}" $1)" == "200"  ]; then
```

Smarter way.

Reply:
using curl changes the http response codes, its no longer 200

I think the function is overkill anyway, this should work
```
if wget -q --spider "$1"; then
```